notebooks/ml-function-anomaly-detect/notebook.ipynb

```diff
@@ -0,0 +1,8872 @@
+          ]
+        }
+      ],
+      "source": [
+        "try:\n",
+        "    # Create a SQLAlchemy engine\n",
+        "    engine = s2.create_engine(database=db_name)\n",
+        "\n",
+        "    # Use pandas.to_sql to load the complete data to a table\n",
+        "    df.to_sql(\"metrics\", con=engine, if_exists='append', index=False)\n",
+        "\n",
+        "    print(\"Data loading complete.\")\n",
+        "except Exception as e:\n",
+        "    print(f\"Error loading data: {e}\")"
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "id": "1a538d2f",
+      "metadata": {},
+      "source": [
+        "## Create Train Test splits from complete data\n",
+        "\n",
+        "Ensure that the rows belonging to the same sequence are grouped in the same split. The following SQL query takes care of this split"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "id": "fc675942",
+      "metadata": {},
+      "outputs": [
+        {
+          "data": {
+            "text/html": [
+              "<span style=\"color: green\">2009 rows affected.</span>"
+            ],
+            "text/plain": [
+              "2009 rows affected."
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        },
+        {
+          "data": {
+            "text/html": [
+              "<span style=\"color: green\">503 rows affected.</span>"
+            ],
+            "text/plain": [
+              "503 rows affected."
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        },
+        {
+          "data": {
+            "text/html": [
+              "<div>\n",
+              "<style scoped>\n",
+              "    .dataframe tbody tr th:only-of-type {\n",
+              "        vertical-align: middle;\n",
+              "    }\n",
+              "\n",
+              "    .dataframe tbody tr th {\n",
+              "        vertical-align: top;\n",
+              "    }\n",
+              "\n",
+              "    .dataframe thead th {\n",
+              "        text-align: right;\n",
+              "    }\n",
+              "</style>\n",
+              "<table border=\"1\" class=\"dataframe\">\n",
+              "  <thead>\n",
+              "    <tr style=\"text-align: right;\">\n",
+              "      <th></th>\n",
+              "    </tr>\n",
+              "  </thead>\n",
+              "  <tbody>\n",
+              "  </tbody>\n",
+              "</table>\n",
+              "</div>"
+            ],
+            "text/plain": [
+              "Empty DataFrame\n",
+              "Columns: []\n",
+              "Index: []"
+            ]
+          },
+          "execution_count": 8,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "%%sql\n",
+        "USE \"temp\";\n",
```

Comment:
If you are going to do this instead of selecting the database in the UI, you should at least use `{{ db_name }}` so that if that variable is changed earlier in the notebook, this still matches.